addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1682,6 +1704,28 @@ export class OdooEditor extends EventTarget {
         this._currentStep.mutations.unshift(...this._historyStashedMutations);
         this._historyStashedMutations = [];
     }
+    /**
+     * Generates the path to a node as an array of indices, relative to a given ancestor.
+     *
+     * @param {Node} node - The node to trace the path for.
+     * @returns {number[]} The path as an array of child indices.
+     */
+    _getNodeIndexPath(node) {
+        return [node, ...ancestors(node, this.editable)].map((ancestor) =>
+            childNodeIndex(ancestor)
+        );
+    }
+    /**
+     * Finds a node in the DOM based on a path of child indices.
+     *
+     * @param {number[]} indexPath - The path as an array of child indices.
+     * @returns {Node|null} The node at the specified path, or null if not found.
+     */
+    _getNodeFromIndexPath(indexPath) {
+        return (
+            indexPath.reduceRight((node, index) => node?.childNodes?.[index], this.editable.parentElement) || null
```

Comment:
Why `|| null`? Without it the return value of `_getNodeFromIndexPath` will be `undefined`, which is perfectly fine IMO. Or did I miss something? For the typing, in TypeScript I'd use `Node?`, but I don't know if it's valid for jsdoc or if it requires to write `undefined` in there specifically.